src/TLuaInterpreter.cpp

```diff
@@ -1857,77 +1857,501 @@ int TLuaInterpreter::getColumnNumber(lua_State* L)
+        lua_pushfstring(L, "startStopWatch: bad argument #1 type (stopwatch id as number or name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    Host& host = getHostFromLua(L);
+    if (lua_type(L, 1) == LUA_TNUMBER) {
+        QPair<bool, QString> result = host.resetStopWatch(lua_tointeger(L, 1));
+        if (!result.first) {
+            lua_pushnil(L);
+            lua_pushstring(L, result.second.toUtf8().constData());
+            return 2;
+        }
+
+        lua_pushboolean(L, true);
         return 1;
+    }
+
+    QPair<bool, QString> result = host.resetStopWatch(QString::fromUtf8(lua_tostring(L, 1)));
+    if (!result.first) {
+        lua_pushnil(L);
+        lua_pushstring(L, result.second.toUtf8().constData());
+        return 2;
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#adjustStopWatch
+int TLuaInterpreter::adjustStopWatch(lua_State* L)
+{
+    if (!(lua_isnumber(L, 1) || lua_isstring(L, 1))) {
+        lua_pushfstring(L, "adjustStopWatch: bad argument #1 type (stopwatch id as number or name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    int watchId = 0;
+    Host& host = getHostFromLua(L);
+    if (lua_type(L, 1) == LUA_TNUMBER) {
+        watchId = lua_tointeger(L, 1);
+    } else {
+        QString name = QString::fromUtf8(lua_tostring(L, 1));
+        // Using an empty string will return the first unnamed stopwatch:
+        watchId = host.findStopWatchId(name);
+        if (!watchId) {
+            lua_pushnil(L);
+            if (name.isEmpty()) {
+                lua_pushstring(L, "no unnamed stopwatches found");
+            } else {
+                lua_pushfstring(L, "stopwatch with name \"%s\" not found", name.toUtf8().constData());
+            }
+            return 2;
+        }
+    }
+
+    if (!lua_isnumber(L, 2)) {
+        lua_pushfstring(L, "adjustStopWatch: bad argument #2 type (modification in seconds as number expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+
+    double adjustment = lua_tonumber(L, 2);
+    bool result = host.adjustStopWatch(watchId, qRound(adjustment * 1000.0));
+    // This is only likely to fail when a numeric first argument was given:
+    if (!result) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "stopwatch with id %d not found", watchId);
+        return 2;
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#deleteStopWatch
+int TLuaInterpreter::deleteStopWatch(lua_State* L)
+{
+    if (!(lua_isnumber(L, 1) || lua_isstring(L, 1))) {
+        lua_pushfstring(L, "deleteStopWatch: bad argument #1 type (stopwatch id as number or name as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    qint64 watchId = 0;
```

Comment:
`lua_tointeger(...)` returns a `lua_Integer` which, according to the documentation: "By default it is a `ptrdiff_t`, which is usually the largest signed integral type the machine handles 'comfortably'". On Windows (like other platforms) this is a 64-bit value but `int`s are only 32 bits on that platform and it thus provokes a warning - and this is one way to remove that.

Reply:
This is once more a question of consistency. Why do other functions work with normal `int`s then? I am not questioning the correctness since I have no experience. But looking at the code I see two versions and wonder "why?"

Reply:
If you have the *clang-code-model* plugin enabled then the `TLuaInterpreter.cpp` gets lit up like a Christmas Tree! This is one source of it being irritated with our code and I was trying to not add to the bonfire.  As it happens it is also silenced by the quick and dirty use of `static_cast<int>(`...`)` around the `lua_tointeger(...)` calls - which is the approach I have taken with a fresh commit in conjunction with switching back to `int`. After all, we are not likely to have the user requesting 2^32 stopwatches - or if we do - they deserve whatever *[nasal daemons](https://www.urbandictionary.com/define.php?term=nasal%20demons)* they get.